### PR TITLE
doc: add jenkins guide

### PR DESCRIPTION
**Pull request by maclover7**
I am VERY happy to get all of this out of my head! Bash commands galore
with information on how to debug and manage our Jenkins clusters.

### CODE REVIEW

File: doc/jenkins-guide.md [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+Issues with the Jenkins clusters are usually reported to either the
+[#node-build IRC channel](irc://irc.freenode.net/node-build), or to the
+[`nodejs/build` issue tracker](https://github.com/nodejs/build/issues).
+
+When trying to fix a worker, ensure that you "mark the node as offline,"
+via the Jenkins worker configure UI, so more failures don't pile up.
+Once you are done fixing the worker, ensure that you return the worker
+to the "online" status.
+
+The most common issues facing workers are explain below, with potential
```
Comment (Trott) [inline doc/jenkins-guide.md]: nit: `explain` -> `explained`

File: doc/jenkins-guide.md [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+
+Overall memory utilization can be found using the `free` command on most
+workers, or the `swap -s -h` command on SmartOS workers.
+
+##### Out of space
+
+First, get statistics on how full (or not) the machine is by running the
+`df -h` command.
+
+If the `Use%` columns appears very high for the worker's largest disk,
```
Comment (Trott) [inline doc/jenkins-guide.md]: `column`?

File: doc/jenkins-guide.md [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+##### IDK what to do
+
+In case the above steps did not work, or you are unsure of what to try,
+this section is for you.
+
+The first thing to remember is that, ultimately, all workers can be
+replaced with newly provisioned ones, so don't worry too much about
+messing up a worker.
+
+The safest bet when dealing with a erroring worker is to re-run its
```
Comment (Trott) [inline doc/jenkins-guide.md]: `an erroring`

File: doc/jenkins-guide.md [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+know the `systemd` service name, say `jenkins-test-digitalocean-ubuntu1804_container-x64-1.service`, for instance:
+
+- To view a list of all active services, you can run
+`systemctl list-units | grep jenkins`.
+- To view the logs for a service, you can run `journalctl -n 50 -u systemctl restart jenkins-test-digitalocean-ubuntu1804_container-x64-1.service | less`
+- To restart a Docker container, restart the associated `systemd` service, for
+example:
+`systemctl restart jenkins-test-digitalocean-ubuntu1804_container-x64-1.service `.
+
+To "SSH into" a Docker container, first you need to find the container
```
Comment (Trott) [inline doc/jenkins-guide.md]: Micro-nit: Remove the quotation marks.

**Comment by Trott** [comment]
Left some nits. If you want me to commit them directly to the branch so you don't have to bother, let me know.

**Comment by maclover7** [comment]
@Trott updated, PTAL

**Comment by maclover7** [comment]
Landed in 839fc90b2929a2fc9e4222ec693c535c500c3ef3

**Pull request merged**